### PR TITLE
backport fixes [REVPI-1937]

### PR DESCRIPTION
**Pull request by iluminat23**

### CODE REVIEW

File: revpi_flat.c [hunk 1]
```diff
@@ -195,20 +196,21 @@ static int revpi_flat_poll_ain(void *data)
...
-			} else {
-				image->drv.cpu_temp = temperature / 1000;
-			}
 		}
 		/* read cpu frequency */
-		image->drv.cpu_freq = bcm2835_cpufreq_get_clock() / 10;
+		freq = bcm2835_cpufreq_get_clock();
+
+		my_rt_mutex_lock(&piDev_g.lockPI);
+		if (piDev_g.thermal_zone != NULL && ret)
```
Comment (linosanfilippo-kunbus) [inline revpi_flat.c:211]: This should be if (piDev_g.thermal_zone != NULL && !ret)

Comment (linosanfilippo-kunbus) [reply revpi_flat.c:211]: Furthermore although correct from the operator priorities I would rather make it explicit: ((piDev_g.thermal_zone != NULL) && !ret) for readability

File: revpi_compact.c [hunk 2]
```diff
@@ -317,22 +317,24 @@ static int revpi_compact_poll_ain(void *data)
...
-				} else {
-					image->drv.i8uCPUTemperature = temp / 1000;
-				}
 			}
 
-			image->drv.i8uCPUFrequency = bcm2835_cpufreq_get_clock() / 10;
+			freq = bcm2835_cpufreq_get_clock();
+
+			my_rt_mutex_lock(&piDev_g.lockPI);
+			if (piDev_g.thermal_zone != NULL && !ret)
```
Comment (linosanfilippo-kunbus) [inline revpi_compact.c:335]: Also here I would rather make it explicit: ((piDev_g.thermal_zone != NULL) && !ret) for readability

**Review by linosanfilippo-kunbus — CHANGES_REQUESTED** [review]

File: revpi_flat.c (hunk 1 above)
Comment (iluminat23) [reply revpi_flat.c]: Done!

**Review by linosanfilippo-kunbus — APPROVED** [review]

**Pull request merged**